servlet/src/main/scala/org/http4s/servlet/ServletIo.scala

```diff
@@ -191,7 +191,7 @@ final case class NonBlockingServletIo[F[_]: Effect](chunkSize: Int) extends Serv
         state.getAndSet(Ready) match {
           case Blocked(cb) => cb(writeChunk)
           case AwaitingLastWrite(cb) => cb(Right(()))
-          case old =>
+          case old @ _ =>
```

Comment:
I'm not sure I like the implicit invocation of unit here.

Reply:
Do you mean there should be a `()` on the right hand side?  The compiler didn't bring it up, but I like that idea.

Reply:
Yes, that's what I mean, despite it working, I find that it's compiler magic rather than pure functions.